src/panels/config/script/ha-script-editor.ts

```diff
@@ -753,25 +760,33 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
       });
       return;
     }
-    const id = this.scriptEntityId
-      ? computeObjectId(this.scriptEntityId)
-      : this._entityId || Date.now();
-    this.hass!.callApi("POST", "config/script/config/" + id, this._config).then(
-      () => {
-        this._dirty = false;
-
-        if (!this.scriptEntityId) {
-          navigate(`/config/script/edit/${id}`, { replace: true });
-        }
-      },
-      (errors) => {
-        this._errors = errors.body.message || errors.error || errors.body;
-        showToast(this, {
-          message: errors.body.message || errors.error || errors.body,
-        });
-        throw errors;
-      }
-    );
+
+    const entry = this.scriptEntityId
+      ? await getExtendedEntityRegistryEntry(
+          this.hass,
+          this.scriptEntityId as string
+        )
+      : undefined;
+    const id = entry ? entry.unique_id : this._entityId || Date.now();
+    try {
+      await this.hass!.callApi(
+        "POST",
+        "config/script/config/" + id,
+        this._config
+      );
+    } catch (errors: any) {
+      this._errors = errors.body.message || errors.error || errors.body;
+      showToast(this, {
+        message: errors.body.message || errors.error || errors.body,
+      });
+      throw errors;
+    }
+
+    this._dirty = false;
+
+    if (!this.scriptEntityId) {
+      navigate(`/config/script/edit/script.${id}`, { replace: true });
```

Comment:
This leads to an error because the entity is not in the registry when the post response is ok (when we call navigate). The only thing I saw it's set timeout but it's ugly... 😞